Mathlib/Topology/Algebra/RestrictedProduct.lean

```diff
@@ -268,6 +282,120 @@ instance [Π i, Ring (R i)] [∀ i, SubringClass (S i) (R i)] :
 
 end Algebra
 
+section eval
+
+variable {S : ι → Type*}
+variable [Π i, SetLike (S i) (R i)]
+variable {B : Π i, S i}
+
+/-- `RestrictedProduct.eval j` is the function from the restricted
+product `Πʳ i, [R i, B i]_[𝓕]` to the
+component `R j`. See also `evalMonoidHom`, `evalAddMonoidHom` and `evalRingHom` for
+evaluation as a monoid or ring homomorphism when `R i` is a monoid or ring.
+-/
+def eval (j : ι) (x : Πʳ i, [R i, B i]_[𝓕]) : R j := x j
```

Comment:
Is the intended simp normal form `eval i x` or `x i`? If it is the former, then this definition should be moved earlier and all previous mentions of `x i` in statements should be replaced. If it is the latter, I'd argue this def shouldn't exist.

Reply:
I've deleted it, I only introduced it because I needed the ring hom.